### ISSUE TITLE
[1.10.2] Diamond Ores very rare or not working

### ISSUE THREAD

**Issue by 016Nojr** [rwtema/extrautilities_old_issues#1146]
I have a Quantum Quarry quarrying away and it has mined about 250000 blocks now, and there is not a single diamond ore mined. Is Diamond Ores working with the quarry? Or is it just SUPER rare?

**Comment by ghost** [comment]
I just ran the quarry for 8.5k blocks and got 6 diamonds.

The quarry generates several chunks of a random biome in another dimension, quarries them and repeats. I don't know if or when it changes biomes, or if specialized ore spawns don't work properly for that dimension. Please post a mod list. In the mean time, try tearing it down and rebuilding it somewhere else.

**Comment by 016Nojr** [comment]
Just using the DW20 pack, version 1.0.0. (ExUtils: alpha-1.2.2)

Just to compare, what dimension number do you have? I have -9999

**Comment by ghost** [comment]
Yes, -9999 is the default dimension. One of my games is on the most recent DW pack and I've found diamonds in play. I think DW did too. However, I set up a quantum quarry in a test world and came up empty after a few thousand blocks. 

I've set up a few and left them running. In the previous DW pack there was a custom ore spawn, but I don't think there's any here. I'll see what I can find for you.

Oh, and don't try accelerating it. I added ProjectE to speed up my tests and the watch of flowing time made the quarry explode. 😃

-----

Update: I've mined a combined ~40k+ from six biomes and the mesa biome showed some easily then stopped, the flower forest showed some after ~16k blocks. The biome changes occasionally, but, if you break the core and replace it, a new biome will be chosen. Maybe you'll have better luck.

-----

Savanna biomes are producing now. I don't know if there's an issue with some biomes, or what. I'd suggest updating to the most recent DW20 pack, 1.1.3, which has the EU 1.1.2 release.

**Comment by 016Nojr** [comment]
> Oh, and don't try accelerating it. I added ProjectE to speed up my tests and the watch of flowing time made the quarry explode. <

Lovely, thanks for checking it out! :)

**Comment by ghost** [comment]
You should be getting around 3 diamonds per 16-18k blocks, assuming vanilla ore gen and no upgrades. If that's what you get with the most recent version, go ahead and close this out, please.

**Comment by 016Nojr** [comment]
Could the issue be that it's on a server?

And regarding your last comment, the staff is claiming that nothing has been changed.

I had no upgrades for the first 200k blocks, after that an Efficiency 5 book. Always had a filter ofc.

**Comment by ghost** [comment]
The version of EU you're using is an alpha from over a month ago. The current release version is 4 releases more recent. True, the only mention of the quarry in the change logs has to do with mob spawning, but you never know what got left out or tweaked under the hood.

Besides there's all sorts of new stuff in it.

I just had a thought, are you using a filter in the quarry?

**Comment by 016Nojr** [comment]
> Always had a filter ofc.

> I just had a thought, are you using a filter in the quarry?

Yes, I am getting everything I want except for the diamonds.

I think we would need Mr. @rwtema Tema to answer if anything else than mob stuff was done recently with the quarry

**Comment by ghost** [comment]
Ok, if you're getting what you want then you know the filter is reversed (it destroys anything that's not in it).

The amount I got in the current version seems a little low, but not as low as nothing for 250k blocks. I'm pretty sure Tema reads all this, since I've seen problems that are posted here get fixed, but he doesn't appear to answer directly; I'm just helping out unofficially. You can always try one of the esper/discord/irc channels or tweeting at him, but I'd start with that update.

**Comment by 016Nojr** [comment]
Yeah, I cant really tell the owners of the server to update the pack since it isnt recommended yet, but trying to push it since it's 3 weeks old. I will reply back when it's updated.

**Comment by ghost** [comment]
The change logs (https://www.feed-the-beast.com/projects/ftb-presents-direwolf20-1-10/files click on the "+1 more" button) show a *lot* of bug fixes since the pack was released.

Just keep in mind, that any fix here would require an update on the owner's end. Be patient, a good guest, and see if IC2 industrial diamonds can fill in for now. Oh, and double check that filter. Good luck!

**Comment by 016Nojr** [comment]
Just to mention it, it just passed 315000 blocks.

And when I wrote the list I just realised that diamonds are not the only thing missing, I am also missing Lapis, Redstone and other modded ores. Only gotten 2 emeralds or something like that aswell.

Could it be that the ores that doesnt drop ores were broken?

And filters cant be the cause because diamond ores are not the only thing in the filter.

I have 3 filters in 1

Filter 1:

- Oak Wood
- Seeds
- Cocoa Beans
- Draconium Ore
- Nether Quartz Ore (Dont ask why)

Filter 2:

- Clay (Missing)
- Amber Ore (Missing)
- Redstone Ore (Missing)
- Lead ore
- Emerald Ore (Missing)
- Diamond Ore (Missing)
- Lapis Ore
- Gold Ore
- Black Quartz (Actually Additions)

Filter 3:

- Bauxite Ore (IE)
- Silver Ore (IE)
- Certus Quartz Ore (AE2) (Missing)
- Tin Ore (IC2)
- Copper Ore (IC2)
- Coal Ore (Missing)
- Iron Ore
- Uranium Ore (IC2)
- Resonating Ore (Deep Resonance)

**Comment by ghost** [comment]
Sorry, there is no config for the ores. Previous versions of the DW20 modpack had mods that customized the ore distribution.

**Comment by 016Nojr** [comment]
Yeah, guess we have to wait for the server then or potentially any news from Mr. Tema

**Comment by 016Nojr** [comment]
Server updated to 1.1.4. Quarry mined 35k blocks. No diamonds found, in addition to redstone, coal, lapis and emerald. Anything wrong with ores not dropping the "non-ore" version of the ore?

**Comment by ghost** [comment]
Not that I can see. I set up a quarry on a fresh world and had six diamonds before I had hit 4k blocks. Is anybody else on the server having this problem?

I have two suggestions. First, tear down and rebuild the setup. Maybe something's wonky and breaking the core will let you start fresh, especially with an updated mod. Second, try switching to a traditional filtering system for a bit. Pipe from the chest using EIO conduits and filters and trash unwanted things with that. Maybe, for some strange reason, diamonds are coming in, but getting filtered out at the quarry. That's a very different bug.

**Comment by 016Nojr** [comment]
It was rebuilt after the update, but I will try collecting everything when I get home from school ~6 hours from now.

**Comment by 016Nojr** [comment]
I feel a bit like a derp now, The Quarry matches with the drop from the block, not the mined block. So for me to get diamonds I had to put in a diamond in a filter, not diamond ore.

Maybe add something to the end of the tooltip which says "(The filter has to match what the block broken drops) @rwtema 

Just close this when you have read it Mr. Tema

**Comment by ghost** [comment]
I missed it too. I made the filter a blacklist and put dirt, etc. in it instead of duplicating the system you had. We live and learn.

**Comment by 016Nojr** [comment]
> I made the filter a blacklist and put dirt, etc. in it instead of duplicating the system you had. We live and learn.

Yep
